### DISCUSSION TITLE
How to use qmol type in PgSQL cartridge similarity search?

### DISCUSSION THREAD

**Discussion by inc0** [rdkit/rdkit#5073] [Q&A] [answered]
Hello!

I'm trying to setup cartridge from chembl to allow similarity search. Thing is, way I'm doing it is to locally create `rdk` schema and populate it according to [this guide](https://www.rdkit.org/docs/Cartridge.html#loading-chembl), then `pg_dump` this schema and restore it on multiple replicas for horizontal scaling.

While restoring of I'm hitting this issue https://github.com/rdkit/rdkit/issues/4442. Following down this discussion I've found [PR](https://github.com/rdkit/rdkit/pull/4448) that seems to suggest that `qmol` is better type to use. Thing is, I can't find alternative to `torsionbv_fp` that would accept `qmol` type. Is there alternative way to calculate these fingerprints from qmol?

**Comment by inc0** [comment]
After a bit more digging it looks like during load of chembl (following the guide), for some reason incorrect SMLIES are generated by `mol_from_ctab` - this is actual error I'm getting while restoring database dump from chembl `could not create molecule from SMILES 'Cc1ccc2c(c1)-n1-c(=O)/c=c\c(=O)-n-2-c2cc(C)ccc2-1' at character 30`

**Comment by inc0** [comment, marked as answer]
Also, found an issue. Turns out it's not RDKit - it's Chembl. Molecule in chembl 29 with molregno `2220555` is just incorrect. It renders into faulty SMILES, even when used with `mol_from_ctab`. Seems like this is only example of such molecule in database, so removing it seems to fix the issue.

**Reply by greglandrum** [reply]
It turns out that this actually is an RDKit bug since the ChEMBL team uses the RDKit to generate canonical SMILES and there is definitely a problem with that one. I will look into it.

**Reply by inc0** [reply]
It caused problems with both psql function from ctab and canonical smiles column directly. That was only record in chembl that caused issues for me

**Reply by inc0** [reply]
It may be worth checking if adding validation of resulting SMILES to functions like `mol_from_ctab` is feasible. That would caught error much quicker.

**Comment by greglandrum** [comment]
> 
> While restoring of I'm hitting this issue #4442. Following down this discussion I've found [PR](https://github.com/rdkit/rdkit/pull/4448) that seems to suggest that `qmol` is better type to use. Thing is, I can't find alternative to `torsionbv_fp` that would accept `qmol` type. Is there alternative way to calculate these fingerprints from qmol?

Just a general point: `qmol` is really intended for use in substructure queries, so it probably wouldn't be appropriate for what you're trying to do here.
